### PR TITLE
add list<> notation in stubs

### PR DESCRIPTION
**Pull request by orklah**
As discussed here https://github.com/vimeo/psalm/issues/2534, this PR use the list<> notation where applicable.

I used the list that @TysonAndre posted here https://github.com/vimeo/psalm/issues/2534#issuecomment-569805992. I made some adjustments when types seemed more detailed in Psalm to avoid erasing good data or when I was not sure a particular notation was interpreted by Psalm.

Judging by the number of `array<int, ...` and `type[]` left in stubs, I think I'll create a second PR later if this one goes well.

### CODE REVIEW

**Comment by orklah** [comment]
The list I used contained a `list<>` type in the parameters of call_user_func_array. While it's coherent with [docs](https://www.php.net/manual/en/function.call-user-func-array.php)
>param_arr
The parameters to be passed to the callback, as an indexed array.

It also works with string indexed arrays:
https://3v4l.org/Xhqmo

IMHO, return types should be as precise as we can (i.e. `list<>` instead of `array<int, ...>`) but params types should be as broad as possible if it's not provoking errors or unexpected behaviour. I think we should ignore the suggested param type for call_user_func_array and call_user_method_array. Any objection?

File: src/Psalm/Internal/CallMap.php [hunk 1]
```diff
@@ -357,7 +357,7 @@
 'ArithmeticError::getTrace' => ['array<int,array<string,mixed>>'],
 'ArithmeticError::getTraceAsString' => ['string'],
 'array_change_key_case' => ['array|false', 'input'=>'array', 'case='=>'int'],
-'array_chunk' => ['array[]', 'input'=>'array', 'size'=>'int', 'preserve_keys='=>'bool'],
```
Comment (TysonAndre) [inline src/Psalm/Internal/CallMap.php]: I missed the $preserve_keys=true case when adding this to Phan. For psalm, two separate signatures may make sense

Comment (orklah) [reply src/Psalm/Internal/CallMap.php]: Good catch. I'll revert for now.

**Comment by TysonAndre** [comment]
> It also works with string indexed arrays:

I'm aware of this - Some of the use of associative arrays or string indexed arrays might be valid and false positives in Phan, but others might be bugs (e.g. an analyzed program where call_user_func should have been used instead of call_user_func_array). Both approaches have arguments for them.

File: src/Psalm/Internal/CallMap.php (hunk 1 above)
Comment (TysonAndre) [reply src/Psalm/Internal/CallMap.php]: I forgot what array_chunk was doing, the original version shouldn't have needed changes. The **inner arrays** are lists (`list<list<...>>`) if preserve_keys is false, but they're not lists if preserve_keys is true.

```
php > var_export(array_chunk([1,2,3], 2, true));
array (
  0 => 
  array (
    0 => 1,
    1 => 2,
  ),
  1 => 
  array (
    2 => 3,  // not a list
  ),
)
```

**Comment by orklah** [comment]
@muglug what's your opinion about https://github.com/vimeo/psalm/pull/2675#issuecomment-577334221 ?
I guess it'll be up to you to decide what you want :)

**Comment by muglug** [comment]
@orklah I’d rather not break existing legacy code (`call_user_func_array`) but the other changes are good

**Comment by muglug** [comment]
Thanks!

**Pull request merged**